drivers/scheduler/k8s/k8s.go

```diff
@@ -199,14 +204,36 @@ func (k *K8s) ParseSpecs(specDir string) ([]interface{}, error) {
 
 	var specs []interface{}
 
+	splitPath := strings.Split(specDir, "/")
+	appName := ""
+	if len(splitPath) > 0 {
```

Comment:
As long as `specDir` is a non-empty string, `len(splitPath)` will always be greater than 0.
If specDir has no slashes in it, `len(splitPath)` will be 1.

Will there ever be a case where `len(splitPath)` <= 0, and be valid for this function?

Reply:
yeah, that's exactly the case, we want to specDir is non-empty implicitly

Reply:
If there is no case where specDir can be empty, then I would suggest changing the logic something like:

```
if len(specDir) == 0 {
    return an err here
}
splitPath := strings.Split(specDir, "/")
appName = splitPath[len(splitPath)-1]
```

Reply:
checked the code above, os.Walk will fail in case specDir 1 and 4, so I removed the check.
case 2 and 3 is expected since the app name will be always the last part of path or it will be specDir itself in case it's not inside another dir.